event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/DefaultSlabAllocator.java

```diff
@@ -73,21 +74,24 @@ private static ScheduledExecutorService defaultExecutor(LifeCycleRegistry lifeCy
         return executor;
     }
 
-    // Compute how many slots in a slab will be used to do a default (new slab) allocation
-    private static int defaultAllocationCount(PeekingIterator<Integer> eventSizes) {
-        int count = 0;
-        int sum = 0;
+    // Compute how many slots in a slab will be used to do a default (new slab) allocation, and how many bytes it will consume
+    static Pair<Integer, Integer> defaultAllocationCount(Pair<Integer,Integer> amountOfSlabUsed, PeekingIterator<Integer> eventSizes) {
```

Comment:
The use of `Pair` makes this sort of confusing. I think a simple pojo may be better.

Reply:
OK I disagree I like the pair better. I am planning to leave it as is, this ticket is on the way to becoming eternal ;-)